### PR TITLE
ORLP-173

### PR DESCRIPTION
**Pull request by Tamara20**
Get one deck, get list of decks, update deck, add deck methods for admin in deck controller

### CODE REVIEW

File: src/main/java/com/softserve/academy/spaced/repetition/repository/CourseRepository.java [hunk 1]
```diff
@@ -26,4 +26,7 @@
     List<Course> findAllByPublishedTrueOrderByRatingDesc();
 
     List<Course> findAllByPublishedTrue();
+
+
+    List<Course> findAll();
```
Comment (b0hdan) [inline src/main/java/com/softserve/academy/spaced/repetition/repository/CourseRepository.java]: It isn't neccessary to write this method here because it is inherited.

File: src/main/java/com/softserve/academy/spaced/repetition/repository/DeckRepository.java [hunk 2]
```diff
@@ -34,4 +36,14 @@
...
     @Query(value = "SELECT d FROM Deck d INNER JOIN d.deckOwner u  WHERE d.id = :deck_id AND u.id = :user_id")
     Deck getDeckByItsIdAndOwnerOfDeck(@Param("deck_id") Long deck_id, @Param("user_id") Long user_id);
+
+
+     //List<Course> getAllCoursesByDeckId( Long deck_id);
+
+   // @Query(value = "DELETE  FROM Course.decks where Course.decks.id = :deck_id")
+   // @Query(value = "update Course c set c.definition = null where c.definition.id = :defId")
+  //  List<Course> deleteDeckFromCourseById(@Param("deck_id") Long deck_id);
+
```
Comment (b0hdan) [inline src/main/java/com/softserve/academy/spaced/repetition/repository/DeckRepository.java]: Delete these comments if you don't use this code.

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 3]
```diff
@@ -68,7 +68,22 @@ public void updateDeck(Deck deck, Long deck_id) {
         deckRepository.save(deck);
     }
 
+
+    @Transactional
     public void deleteDeck(Long deck_id) {
```
Comment (b0hdan) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: Rename the variable in camelCase.

File: src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java [hunk 4]
```diff
@@ -131,9 +128,39 @@ public void updateDeck(@RequestBody Deck deck, @PathVariable Long deck_id) {
...
-        deckService.deleteDeck(deck_id);
+
+    @Auditable(action = AuditingAction.VIEW_DECKS_ADMIN)
+    @GetMapping(value = "/api/admin/decks")
+    public ResponseEntity<List<DeckOfUserManagedByAdminDTO>> getAllDecksForAdmin() {
+        List<Deck> decksList = deckService.getAllOrderedDecks();
+        Link collectionLink = linkTo(methodOn(DeckController.class).getAllDecksForAdmin()).withSelfRel();
+        List<DeckOfUserManagedByAdminDTO> decks = DTOBuilder.buildDtoListForCollection(decksList,
+                DeckOfUserManagedByAdminDTO.class, collectionLink);
+        return new ResponseEntity<>(decks, HttpStatus.OK);
```
[the remark this answers is not in the document]
Comment (Tamara20) [reply src/main/java/com/softserve/academy/spaced/repetition/controller/DeckController.java]: All controllers are like this controller, so it will be a new task .

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 5]
```diff
@@ -86,4 +91,14 @@ public void deleteDeck(Long deckId) {
...
 
+    public void createNewDeck(Deck newDeck, Long category_id) throws NotAuthorisedUserException {
+        User user = userService.getAuthorizedUser();
+        Deck deck = new Deck();
+        deck.setName(newDeck.getName());
+        deck.setDescription(newDeck.getDescription());
+        deck.setCategory(categoryRepository.findById(category_id));
+        deck.setDeckOwner(user);
+        Deck save = deckRepository.save(deck);
+        newDeck.setId(save.getId());
```
Comment (b0hdan) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: Why can't you just use newDeck instead of creating a new object? I don't say it's wrong, just asking.

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 6]
```diff
@@ -25,6 +24,12 @@
     @Autowired
     private CourseRepository courseRepository;
 
+    @Autowired
+    private UserService userService;
+
+    @Autowired
+    private UserRepository userRepository;
```
Comment (b0hdan) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: This field is never used, isn't it?

File: src/main/java/com/softserve/academy/spaced/repetition/logger/logs/log.txt [hunk 7]
```diff
@@ -0,0 +1 @@
+2017-09-21 23:45:23,998 [WARN ][    logger.Log4jImpl] - Exception in controller = com.softserve.academy.spaced.repetition.controller.DeckController@169f19c1, method = void com.softserve.academy.spaced.repetition.controller.DeckController.deleteDeckForAdmin(Long)
```
Comment (b0hdan) [inline src/main/java/com/softserve/academy/spaced/repetition/logger/logs/log.txt]: revert this changes

File: src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java [hunk 8]
```diff
@@ -41,34 +42,56 @@
...
+        deck.setId(deckId);
         deckRepository.save(deck);
     }
 
-    public void deleteDeck(Long deck_id) {
-        deckRepository.delete(deck_id);
+
+    @Transactional
+    public void deleteDeck(Long deckId) {
+        List<Course> courses = courseRepository.findAll();
```
Comment (antonbakalets) [inline src/main/java/com/softserve/academy/spaced/repetition/service/DeckService.java]: This method loads to much data before removing one deck.
